codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationHandlerGenerator.kt

```diff
@@ -78,41 +79,62 @@ class ServerOperationHandlerGenerator(
                 """.trimIndent(),
                 *codegenScope
             ) {
+                // Instrument operation handler at callsite.
+                val operationHandlerInvoke = if (state) {
+                    "self(input_inner, state)"
+                } else {
+                    "self(input_inner)"
+                }
+                val operationHandlerCall =
+                    """
+                    let input_inner = input_wrapper.into();
+                    #{Tracing}::debug!(input = ?input_inner, "calling operation handler");
+                    let output_inner = $operationHandlerInvoke
+                        .instrument(#{Tracing}::debug_span!("${operationName}_handler"))
+                        .await;
+                    #{Tracing}::debug!(output = ?output_inner, "operation handler returned");
+                    """
                 val callImpl = if (state) {
                     """
                     let state = match $serverCrate::extension::extract_extension(&mut req).await {
                         Ok(v) => v,
                         Err(extension_not_found_rejection) => {
+                            #{Tracing}::error!(?extension_not_found_rejection, "unable to extract extension from request; maybe you forgot to register it with `AddExtensionLayer`?");
                             let extension = $serverCrate::extension::RuntimeErrorExtension::new(extension_not_found_rejection.to_string());
                             let runtime_error = $serverCrate::runtime_error::RuntimeError {
                                 protocol: #{SmithyHttpServer}::protocols::Protocol::${protocol.name.toPascalCase()},
                                 kind: extension_not_found_rejection.into(),
                             };
                             let mut response = runtime_error.into_response();
                             response.extensions_mut().insert(extension);
-                            return response.map($serverCrate::body::boxed);
+                            let response = response.map($serverCrate::body::boxed);
+                            #{Tracing}::debug!(?response, "returning HTTP response");
```

Comment:
Response bodies should probably only be logged at `TRACE` level, since they can be very big.

Same for request bodies, which I'm not logging anywhere.

Reply:
Most boxed types don't have a real debug impl https://docs.rs/http-body/latest/src/http_body/combinators/box_body.rs.html#32-36 so is this actually useful to print the body?